Eevee.Sleep.Bot/Modules/SlashCommands/RoleManagementSlashModule.cs

```diff
@@ -0,0 +1,69 @@
+using Discord;
+using Discord.Interactions;
+using JetBrains.Annotations;
+using Eevee.Sleep.Bot.Controllers.Mongo;
+using Eevee.Sleep.Bot.Utils;
+
+namespace Eevee.Sleep.Bot.Modules.SlashCommands;
+
+[Group("role", "Commands for managing the roles on a user.")]
+public class RoleManagementSlashModule : InteractionModuleBase<SocketInteractionContext> {
+    [SlashCommand("change", "Select a specific one from the role you have")]
+    [UsedImplicitly]
+    public Task ChangeRoleAsync() {
+        return Context.Interaction.RespondAsync(
+            components: DiscordMessageMaker.MakeRoleSelectButton(
+                hasRoles: DiscordRoleContoller.FindRoleIdsByUserId(Context.User.Id), 
+                guildRoles: Context.Guild.Roles),
+            ephemeral: true
+        );
+    }
+
+    [SlashCommand("remove", "Remove the role ownership of a user")]
+    [RequireOwner]
+    [UsedImplicitly]
+    public async Task RemoveRoleAsync(IUser user, IRole role) {
+        await DiscordRoleContoller.RemoveRoles(user.Id, [role.Id]);
+
+        await Context.Interaction.RespondAsync(
+            "Role record deleted.",
+            ephemeral: true
+        );
+    }
+
+    [SlashCommand("track", "Track the specific role")]
+    [RequireOwner]
+    [UsedImplicitly]
+    public async Task TrackRoleAsync(IRole role) {
+        await Context.Interaction.RespondAsync(
+            "Role record update/create task dispatched.",
+            ephemeral: true
+        );
+
+        var users = Context.Guild.Users;
+        var roleOwnedUsers = users.Where(x => x.Roles.Contains(role));
+
+        foreach (var user in roleOwnedUsers) {
+            await DiscordRoleContoller.AddRoles(user.Id, [role.Id]);
```

Comment:
Have a method to allow 1 DB call instead. You will likely need `updateMany()`

> [!IMPORTANT]
> The basic idea of this (and this applies to every DB call) is that you want to reduce the DB calls as many as possible. Imagine there's role with 1K users to get tracked. Using `updateMany()`, it could be a single call. Using the current implementation, it's 1K requests to the DB.
> Always try to see if the `foreach` call can be omitted. This applies to everywhere.

Reply:
fixed~
[1ca7650](https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/pull/25/commits/1ca7650e92bcf1b8e1267ae7e1e955be955cb8b3)